### ISSUE TITLE
Network and subnet field of google_vpc_access_connector conflict

### ISSUE THREAD

**Issue by red8888** [hashicorp/terraform-provider-google#12249]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Summary

We started seeing a diff in our plans. We've made no changes but maybe a provider was updated.

### Terraform Version
```

Terraform v0.14.4
+ provider registry.terraform.io/hashicorp/aws v3.64.2
+ provider registry.terraform.io/hashicorp/google v3.90.1
+ provider registry.terraform.io/hashicorp/google-beta v3.90.1
+ provider registry.terraform.io/hashicorp/random v3.1.0
```

### Affected Resource(s)

* google_vpc_access_connector

### Terraform Configuration Files

```tf
resource "google_vpc_access_connector" "aaaa" {
  provider       = google-beta
  name           = "connector"
  region         = data.google_compute_subnetwork.shared-vpc.region
  max_throughput = "300"
  subnet {
    name       = data.google_compute_subnetwork.shared-vpc.name
    project_id = data.google_compute_subnetwork.shared-vpc.project
  }
}
```

### Expected Behavior

No diff

### Actual Behavior

Seeing a diff that looks like this. You cannot specify network and subnet, but terraform seems to have a problem with network being null and it wants to destroy and create it. When I do a state show I see no network field

```
-/+ resource "google_vpc_access_connector" "aaa" {
      ~ id             = "projects/service-network/locations/us-central1/connectors/connector" -> (known after apply)
      ~ max_instances  = 3 -> (known after apply)
      ~ min_instances  = 2 -> (known after apply)
        name           = "shared-vpc-connector"
      - network        = "service-network-name" -> null # forces replacement
      ~ project        = "service-project" -> (known after apply)
      ~ self_link      = "projects/service-network/locations/us-central1/connectors/connector" -> (known after apply)
      ~ state          = "READY" -> (known after apply)
        # (4 unchanged attributes hidden)


      - timeouts {}
        # (1 unchanged block hidden)
    }
```

### Crappy workaround
Adding a lifecycle ignore fixes the diff:
`  lifecycle {
    ignore_changes = [network]
  }`

**Comment by edwardmedia** [comment]
@red8888 can you share the debug log for your initial apply?

**Comment by megan07** [comment]
Hi @red8888, this appears to be fixed in the most recent version. It looks like it was fixed by [this PR](https://github.com/GoogleCloudPlatform/magic-modules/pull/5886) and released in version `4.18.0`. Thanks!
